src/qml/qgismobileapp.qml

```diff
@@ -375,106 +379,126 @@ ApplicationWindow {
     }
   }
 
-  Button {
-    id: gpsButton
-    state: positionSource.active ? "On" : "Off"
-    visible: positionSource.valid
-    round: true
-
+  Column {
+    id: mainToolBar
     anchors.left: dashBoard.right
     anchors.leftMargin: 4 * dp
     anchors.top: mainMenuBar.bottom
     anchors.topMargin: 4 * dp
+    spacing: 4*dp
 
-    bgcolor: "#64B5F6"
+    Button {
+      id: gpsButton
+      state: positionSource.active ? "On" : "Off"
+      visible: positionSource.valid
+      round: true
 
-    states: [
-      State {
+      bgcolor: "#64B5F6"
 
-        name: "Off"
-        PropertyChanges {
-          target: gpsButton
-          iconSource: Style.getThemeIcon( "ic_location_disabled_white_24dp" )
-          bgcolor: "lightgrey"
-        }
-      },
+      states: [
+        State {
 
-      State {
-        name: "On"
-        PropertyChanges {
-          target: gpsButton
-          bgcolor: "#64B5F6"
-          iconSource: positionSource.position.latitudeValid ? Style.getThemeIcon( "ic_my_location_white_24dp" ) : Style.getThemeIcon( "ic_gps_not_fixed_white_24dp" )
+          name: "Off"
+          PropertyChanges {
+            target: gpsButton
+            iconSource: Style.getThemeIcon( "ic_location_disabled_white_24dp" )
+            bgcolor: "lightgrey"
+          }
+        },
+
+        State {
+          name: "On"
+          PropertyChanges {
+            target: gpsButton
+            bgcolor: "#64B5F6"
+            iconSource: positionSource.position.latitudeValid ? Style.getThemeIcon( "ic_my_location_white_24dp" ) : Style.getThemeIcon( "ic_gps_not_fixed_white_24dp" )
+          }
         }
-      }
-    ]
+      ]
 
-    onClicked: {
-      console.warn("Centering")
-      if ( positionSource.projectedPosition.x )
-      {
-        console.warn("Centering to " + positionSource.projectedPosition.x + " " + positionSource.projectedPosition.y )
-        mapCanvas.mapSettings.setCenter(positionSource.projectedPosition)
-
-        if ( !positionSource.active )
-        {
-          positionSource.active = true;
-          displayToast( qsTr( "Activating positioning service..." ) )
-        }
-      }
-      else
-      {
-        if ( positionSource.valid )
+      onClicked: {
+        console.warn("Centering")
+        if ( positionSource.projectedPosition.x )
         {
-          if ( positionSource.active )
+          console.warn("Centering to " + positionSource.projectedPosition.x + " " + positionSource.projectedPosition.y )
+          mapCanvas.mapSettings.setCenter(positionSource.projectedPosition)
+
+          if ( !positionSource.active )
           {
-            displayToast( qsTr( "Waiting for location..." ) )
+            positionSource.active = true;
```

Comment:
There's a change in the logic here, it does not check for positionSource validity before activating. Might have an impact?

Reply:
Ah no, it's just indentation that changed...